core/src/main/java/org/apache/iceberg/avro/AvroIterable.java

```diff
@@ -150,7 +153,7 @@ public D next(D reuse) {
       try {
         return reader.next(reuse);
       } catch (IOException e) {
-        throw new RuntimeIOException(e, "Failed to read next record");
+        throw new RuntimeIOException(e, "Failed to read next record at file: %s", file);
       }
```

Comment:
I think we should probably add some checks to the below methods as well, we can Rethrow the IOException but enrich it with a message about which file broke

Reply:
@szehon-ho + @aokolnychyi What do you think?

Reply:
I think I may be talking myself out of this since all those other methods should be called because of the few you fixed already so we should be enriching them anyway ... I think :)